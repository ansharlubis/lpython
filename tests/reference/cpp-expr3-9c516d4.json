{
    "basename": "cpp-expr3-9c516d4",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/expr3.py",
    "infile_hash": "4fbbd9ddebefcc9afdd6fdc17e16313fafc5b3e214595d6ad62c10cb",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-expr3-9c516d4.stdout",
    "stdout_hash": "3efdd3433672504d35c0cdf426e0162b862f9d6e0ac58ec65f787f46",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}