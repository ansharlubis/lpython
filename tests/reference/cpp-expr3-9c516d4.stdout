([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         test_cast: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           b [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mReal[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               test_cast [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mb[39m) 
                  ([1m[35mCast[39m[0m 
                     ([1m[35mRealConstant[39m[0m   4.20000000000000018e+00 
                        ([1m[35mReal[39m[0m 8 [])) [1m[32m
                     RealToReal[39m[0m 
                     ([1m[35mReal[39m[0m 4 []) 
                     ([1m[35mRealConstant[39m[0m   4.20000000000000018e+00 
                        ([1m[35mReal[39m[0m 4 []))) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mRealBinOp[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) ()) [1m[32m
                     Mul[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) 
                     ([1m[35mReal[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mb[39m) 
                  ([1m[35mRealBinOp[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) [1m[32m
                     Add[39m[0m 
                     ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m1[39m 
                        ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) ()) 
                     ([1m[35mReal[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mRealBinOp[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     Sub[39m[0m 
                     ([1m[35mRealConstant[39m[0m   3.89999999999999991e+00 
                        ([1m[35mReal[39m[0m 8 [])) 
                     ([1m[35mReal[39m[0m 8 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) 
                  ([1m[35mRealBinOp[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     Div[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) 
                     ([1m[35mReal[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mb[39m) 
                  ([1m[35mCast[39m[0m 
                     ([1m[35mRealBinOp[39m[0m 
                        ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m3[39m 
                           ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                           IntegerToReal[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                        Div[39m[0m 
                        ([1m[35mCast[39m[0m ([1m[35mIntegerConstant[39m[0m [36m4[39m 
                           ([1m[35mInteger[39m[0m 4 [])) [1m[32m
                           IntegerToReal[39m[0m 
                           ([1m[35mReal[39m[0m 8 []) ()) 
                        ([1m[35mReal[39m[0m 8 []) ()) [1m[32m
                     RealToReal[39m[0m 
                     ([1m[35mReal[39m[0m 4 []) ()) ()) 
               ([1m[35mIf[39m[0m 
                  ([1m[35mRealCompare[39m[0m 
                     ([1m[35mCast[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                        IntegerToReal[39m[0m 
                        ([1m[35mReal[39m[0m 4 []) ()) [1m[32m
                     Lt[39m[0m 
                     ([1m[35mVar[39m[0m 2 [33mb[39m) 
                     ([1m[35mLogical[39m[0m 4 []) ()) [
                  ([1m[35mPrint[39m[0m () [
                     ([1m[35mStringConstant[39m[0m "a < b" 
                        ([1m[35mCharacter[39m[0m 1 5 () []))] () ())] [])] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [])
#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <complex>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

// Forward declarations
void test_cast();
namespace {
}

// Implementations
void test_cast()
{
    int a;
    float b;
    a = 2;
    b =   4.20000000000000018e+00;
    a = (float)(a)*b;
    b = b + (float)(1);
    a = 5;
    a = (float)(a) -   3.89999999999999991e+00;
    a = (float)(a)/b;
    b = (float)(3)/(float)(4);
    if ((float)(a) < b) {
        std::cout << "a < b" << std::endl;
    }
}

namespace {

void main2() {
}

}
int main(int argc, char* argv[])
{
    Kokkos::initialize(argc, argv);
    main2();
    Kokkos::finalize();
    return 0;
}
